src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -702,3 +708,201 @@ void QgsBookmarkLocatorFilter::triggerResult( const QgsLocatorResult &result )
   QModelIndex index = qvariant_cast<QModelIndex>( result.userData );
   QgisApp::instance()->zoomToBookmarkIndex( index );
 }
+
+//
+// QgsGotoLocatorFilter
+//
+
+QgsGotoLocatorFilter::QgsGotoLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{}
+
+QgsGotoLocatorFilter *QgsGotoLocatorFilter::clone() const
+{
+  return new QgsGotoLocatorFilter();
+}
+
+void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  if ( feedback->isCanceled() )
+    return;
+
+  const QgsCoordinateReferenceSystem currentCrs = QgisApp::instance()->mapCanvas()->mapSettings().destinationCrs();
+  const QgsCoordinateReferenceSystem wgs84Crs( QStringLiteral( "EPSG:4326" ) );
+
+  // Coordinates such as 106.8468,-6.3804
+  QRegularExpression separatorRx( QStringLiteral( "^([0-9\\-\\.]*)[\\s\\,]*([0-9\\-\\.]*)$" ) );
+  QRegularExpressionMatch match = separatorRx.match( string.trimmed() );
+  if ( match.hasMatch() )
+  {
+    bool okX = false;
+    bool okY = false;
+    double posX = match.captured( 1 ).toDouble( &okX );
+    double posY = match.captured( 2 ).toDouble( &okY );
```

Comment:
Will this handle different locales correctly or maybe we need something like `QLocale::toDouble`?

Reply:
Locale stuff will come in a follow up PR.